src/vs/base/common/glob.ts

```diff
@@ -264,6 +268,16 @@ export function match(arg1: string | IExpression, path: string, siblings?: strin
 
 	// Glob with String
 	if (typeof arg1 === 'string') {
+
+		if (trivia1.test(arg1)) {
+			// common pattern: **/*.txt just need endsWith check
+			return strings.endsWith(path, arg1.substr(4)); // '**/*'.length === 4
```

Comment:
@outcoldman the tests you were looking for: https://github.com/Microsoft/vscode/blob/master/src/vs/base/test/common/glob.test.ts